### PR TITLE
Improve search experience

### PR DESCRIPTION
**Pull request by joeroe**
In the spirit of https://github.com/xronos-ch/orga/issues/108, adds more information to search results, allows filtering of results by type, and improves the visual appearance.

### CODE REVIEW

File: app/views/searches/_form.html.erb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <form class="input-group mb-3" 
       role="search"
-      action="/search"
+      action="<%= search_action || "/search" %>"
```
Comment (MartinHinz) [inline app/views/searches/_form.html.erb:3]: This does not work for me.

`undefined local variable or method 'search_action' for an instance of #<Class:0x000000017773d520>`

My mistake?

File: app/models/citation.rb [hunk 2]
```diff
@@ -23,4 +23,8 @@ def self.label
     "citation"
   end
 
+  def self.icon
+    "icons/citation.svg"
```
Comment (MartinHinz) [inline app/models/citation.rb:27]: Needed to precompile:assets. Is that intended?

File: app/views/references/_reference.html.erb [hunk 3]
```diff
@@ -1,7 +1,29 @@
...
-  </p>
+<article class="card">
+
+  <div class="card-header">
+    <%= xr_icon Reference, {}, { class: "img-fluid", style: "width: auto; height: 1.1rem;" } %>
+    <%= Reference.label.humanize %>
+  </div>
+
+  <div class="card-body">
+    <h1 class="card-title h4">
```
Comment (MartinHinz) [inline app/views/references/_reference.html.erb:9]: That might be a bit bold at times. Maybe more subtle/smaller?

**Review by MartinHinz — CHANGES_REQUESTED** [review]
I am not sure where search_action is coming from!? Maybe you can fix this? Rest might be only matter of taste!?

File: app/models/citation.rb (hunk 2 above)
Comment (joeroe) [reply app/models/citation.rb:27]: I didn't; are you using `bin/dev`? In any case, it'll happen in production: https://github.com/xronos-ch/xronos.rails/blob/master/entrypoint.sh

File: app/views/references/_reference.html.erb (hunk 3 above)
Comment (joeroe) [reply app/views/references/_reference.html.erb:9]: Yeah, I'd forgotten that in production `short_ref` is often not very short... how about as a quick fix we chop it off at 25 characters or so, pending adding `LONG_SHORT_REF` as an issue and cleaning the data (https://github.com/xronos-ch/xronos.rails/issues/260)?

File: app/views/searches/_form.html.erb (hunk 1 above)
Comment (joeroe) [reply app/views/searches/_form.html.erb]: Removed in the latest commit.

**Review by MartinHinz — APPROVED** [review]
Thanks, now it works. Looking good, font size of the card headers is still a bit massive for my taste, but this is a matter of taste ;-)

Also, a good reminder that I should take care of the changelog in my own PRs!

**Pull request merged**